### PR TITLE
Limit order connector

### PR DESCRIPTION
**Pull request by shriyatyagii**
Limit Order Connector added
Contract- 0x9cAf9279C8Aa3620498236c10e9af08164dF016E
[View in PolygonScan](https://polygonscan.com/address/0x9cAf9279C8Aa3620498236c10e9af08164dF016E#code)

### CODE REVIEW

File: contracts/polygon/connectors/limitOrder/helpers.sol [hunk 1]
```diff
@@ -0,0 +1,86 @@
+pragma solidity ^0.8.1;
+// SPDX-License-Identifier: MIT
+
+import { Basic } from "../../common/basic.sol";
+import "./interface.sol";
+
+contract Helpers is Basic {
+
+    UniLimitOrder limitCon_ = UniLimitOrder(0xfC428E6535dC5Fee30fb57cFc93EBB1D92fdCf6e);
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/helpers.sol]:
```suggestion
    UniLimitOrder public constant limitCon_ = UniLimitOrder(0xfC428E6535dC5Fee30fb57cFc93EBB1D92fdCf6e);
```

Comment (KaymasJain) [reply contracts/polygon/connectors/limitOrder/helpers.sol]: Every variable in connectors should be a constant or immutable

File: contracts/polygon/connectors/limitOrder/helpers.sol [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        {
+            (address token0, ) = sortTokenAddress(
+                address(token0_),
+                address(token1_)
+            );
+
+            if (token0 != address(token0_)) {
+                (token0_, token1_) = (token1_, token0_);
+            }
+        }
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/helpers.sol]: I think we can remove this thing. Let it get checked on the core contracts itself.

File: contracts/polygon/connectors/limitOrder/helpers.sol [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        );
+
+        if(params_.token0to1){
+            amountSend_ = params_.amount == type(uint128).max ? getTokenBal(TokenInterface(params_.token0)) : params_.amount;
+            convertMaticToWmatic(address(token0_) == wmaticAddr, token0_, amountSend_);
+            approve(TokenInterface(token0_), address(limitCon_), amountSend_);
+        } else {
+            amountSend_ = params_.amount == type(uint128).max ? getTokenBal(TokenInterface(params_.token1)) : params_.amount;
+            convertMaticToWmatic(address(token1_) == wmaticAddr, token1_, amountSend_);
+            approve(TokenInterface(token1_), address(limitCon_), amountSend_);
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/helpers.sol]:
```suggestion
            approve(token1_, address(limitCon_), amountSend_);
```

File: contracts/polygon/connectors/limitOrder/helpers.sol [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+        (TokenInterface token0_, TokenInterface token1_) = changeMaticAddress(
+            params_.token0,
+            params_.token1
+        );
+
+        if(params_.token0to1){
+            amountSend_ = params_.amount == type(uint128).max ? getTokenBal(TokenInterface(params_.token0)) : params_.amount;
+            convertMaticToWmatic(address(token0_) == wmaticAddr, token0_, amountSend_);
+            approve(TokenInterface(token0_), address(limitCon_), amountSend_);
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/helpers.sol]:
```suggestion
            approve(token0_, address(limitCon_), amountSend_);
```

File: contracts/polygon/connectors/limitOrder/interface.sol [hunk 5]
```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.8.1;
+// SPDX-License-Identifier: MIT
+
+interface UniLimitOrder {
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/interface.sol]:
```suggestion
interface IUniLimitOrder {
```

Comment (KaymasJain) [reply contracts/polygon/connectors/limitOrder/interface.sol]: Follow a pattern. In the interface always add `I` in beginning or `Interface` in the end. Makes it easy to separate them from contracts. After updating this you will need to update things at many all the places you are using it.

File: contracts/polygon/connectors/limitOrder/main.sol [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import {UniLimitOrder} from "./interface.sol";
+import {TokenInterface} from "../../common/interfaces.sol";
+
+/**
+ * @title LimitOrderConnector.
+ * @dev Connector for Limit Order Swap on Uni V3.
+ */
+contract LimitOrderConnector is Helpers {
+
+    function createPosition(
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/main.sol]:
```suggestion
    function create(
```

File: contracts/polygon/connectors/limitOrder/main.sol [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            tokenId_,
+            liquidity_,
+            minAmount_,
+            params_.tickLower,
+            params_.tickUpper
+        );
+    }
+
+
+    function withdrawMid(
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/main.sol]:
```suggestion
    function closeMid(
```

File: contracts/polygon/connectors/limitOrder/main.sol [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+        setUint(setIds_[0], amount0);
+        setUint(setIds_[1], amount1);
+
+        eventName_ = "LogWithdrawMid(uint256,uint256,uint256,uint256)";
+        eventParam_ = abi.encode(tokenId_, liquidity_, amount0, amount1);
+    }
+
+
+    function withdrawFull(
```
Comment (KaymasJain) [inline contracts/polygon/connectors/limitOrder/main.sol]:
```suggestion
    function closeFull(
```

**Review by KaymasJain — DISMISSED** [review]

File: contracts/polygon/connectors/limitOrder/helpers.sol [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        uint256 amountSend_;
+
+        (TokenInterface token0_, TokenInterface token1_) = changeMaticAddress(
+            params_.token0,
+            params_.token1
+        );
+
+        if(params_.token0to1){
+            amountSend_ = params_.amount == type(uint128).max ? getTokenBal(TokenInterface(params_.token0)) : params_.amount;
+            convertMaticToWmatic(address(token0_) == maticAddr, token0_, amountSend_);
```
Comment (Vaibhav20201) [inline contracts/polygon/connectors/limitOrder/helpers.sol]:
```suggestion
            convertMaticToWmatic(params_.token0 == maticAddr, token0_, amountSend_);
```

File: contracts/polygon/connectors/limitOrder/helpers.sol [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+            approve(token0_, address(limitCon_), amountSend_);
+        } else {
+            amountSend_ = params_.amount == type(uint128).max ? getTokenBal(TokenInterface(params_.token1)) : params_.amount;
+            convertMaticToWmatic(address(token1_) == maticAddr, token1_, amountSend_);
```
Comment (Vaibhav20201) [inline contracts/polygon/connectors/limitOrder/helpers.sol]:
```suggestion
            convertMaticToWmatic(params_.token1 == maticAddr, token1_, amountSend_);
```

File: contracts/polygon/connectors/limitOrder/helpers.sol (hunk 9 above)
Comment (KaymasJain) [reply contracts/polygon/connectors/limitOrder/helpers.sol]: Why is this needed?

**Pull request closed without merging**